### PR TITLE
MLT: Coincidence trigger (trigger deliverable for v4.1.0)

### PR DESCRIPTION
**Pull request by MRiganSUSX**
This is the implementation of coincidence trigger as discussed in
[Trigger: Issue 220](https://github.com/DUNE-DAQ/trigger/issues/220)

### CODE REVIEW

File: plugins/ModuleLevelTrigger.cpp [hunk 1]
```diff
@@ -394,49 +419,63 @@ ModuleLevelTrigger::send_trigger_decisions()
...
-  TLOG_DEBUG(3) << "Override?: " << override_flag;
-  if ((!m_paused.load() && !m_dfo_is_busy.load()) || override_flag) {
+
+  if (m_use_bitwords) {
+    // Check trigger bitwords
+    m_TD_bitword = get_TD_bitword(pending_td);
+    m_bitword_check = check_trigger_bitwords();
+  }
+
+  if ((!m_use_bitwords) || (m_bitword_check)) {
```
Comment (ArturSztuc) [inline plugins/ModuleLevelTrigger.cpp:429]: This is a big if-else. Could easily convert into small one by reversing and return, like so:

```
if( !((!m_use_bitwords) || (m_bitword_check)){
    m_td_not_triggered_count++;
    m_td_not_triggered_tc_count += pending_td.contributing_tcs.size();
    return;
}
```

And the rest of the code in that function (that was above, and now below) here, completely outside large if statement

File: plugins/ModuleLevelTrigger.hpp [hunk 2]
```diff
@@ -137,11 +137,22 @@ class ModuleLevelTrigger : public dunedaq::appfwk::DAQModule
   int m_earliest_tc_index;
   int get_earliest_tc_index(const PendingTD& pending_td);
 
+  // Bitwords logic
+  bool m_use_bitwords;
+  nlohmann::json m_trigger_bitwords_json;
+  bool m_bitword_check;
+  std::bitset<16> m_TD_bitword;
+  std::vector<std::bitset<16>> m_trigger_bitwords;
+  std::bitset<16> get_TD_bitword(const PendingTD& ready_td);
```
Comment (ArturSztuc) [inline plugins/ModuleLevelTrigger.hpp:146]: I'm wondering if the number 16 could be placed in some more more global variable so it's not hard-coded here (and many other places). Probably not configurable?

**Review by ArturSztuc — APPROVED** [review]
Overall looks good. I made a couple of comments/recommendations, but they are minor and not significant -- so I don't want them to stop this PR.

File: plugins/ModuleLevelTrigger.cpp (hunk 1 above)
Comment (MRiganSUSX) [reply plugins/ModuleLevelTrigger.cpp:429]: Thanks for the comment. That would probably look nicer but does not change the logic. I will keep that in mind for future dev but will skip it for now.

File: plugins/ModuleLevelTrigger.hpp (hunk 2 above)
Comment (MRiganSUSX) [reply plugins/ModuleLevelTrigger.hpp:146]: As we discussed offline, this is not straightforward and would likely need dedicated spot somewhere outside daqconf. Let's keep it as is for now and revisit at later point.

**Pull request merged**